metrics-core/src/main/java/com/yammer/metrics/core/MetricName.java

```diff
@@ -205,29 +212,30 @@ public boolean equals(Object o) {
         if (this == o) { return true; }
         if (o == null || getClass() != o.getClass()) { return false; }
         final MetricName that = (MetricName) o;
-        return mBeanName.equals(that.mBeanName);
+        return Objects.equals(group, that.group) &&
+            Objects.equals(type, that.type) &&
+            Objects.equals(name, that.name) &&
+            Objects.equals(scope, that.scope) &&
+            Objects.equals(mBeanName, that.mBeanName) &&
```

Comment:
Should we exclude `mBeanName` from `equals` and `hash`? Otherwise I worry that it could lead to incorrect comparisons.

Reply:
There were test failures when I did that, given that it's _always_ `Optional.empty()` in our codebase it seems pretty harmless to include?

Reply:
Yup, I should have read more closely for the "always" part